x/emissions/module/rewards/rewards_internal.go

```diff
@@ -435,23 +435,27 @@ func GetAllReputersOutput(
 	stakes []alloraMath.Dec,
 	initialCoefficients []alloraMath.Dec,
 	numReputers int64,
-	learningRate alloraMath.Dec,
-	gradientDescentMaxIters uint64,
-	epsilonReputer alloraMath.Dec,
-	epsilon alloraMath.Dec,
-	minStakeFraction alloraMath.Dec,
-	maxGradientThreshold alloraMath.Dec,
+	params types.Params,
+	allCoefficientsZero bool,
 ) ([]alloraMath.Dec, []alloraMath.Dec, error) {
 	coefficients := make([]alloraMath.Dec, len(initialCoefficients))
 	copy(coefficients, initialCoefficients)
 
+	// Check if all coefficients are zero
+	// If so, set all coefficients to the fallback listening coefficient
+	if allCoefficientsZero {
+		for i := range coefficients {
+			coefficients[i] = params.FallbackListeningCoefficient
+		}
+	}
+
```

Comment:
You're splitting up control flow between scopes here and the scope that calls `GetAllReputersOutput()`.

In the calling scope, you check if `allCoefficientsZero`, yet in here you recheck that boolean and actually apply that condition.

What you should instead do is just pass in a conditionally-altered `initialCoefficients`, one that uses fallbacks or not depending on the control flow. This makes the code way more readable.

Reply:
This also affects L79 as of this current commits.